File: #0435.non-overlapping-intervals.cpp
class Solution {
public:
    int eraseOverlapIntervals(vector<vector<int>>& intervals) {
        
        std::sort(intervals.begin(), intervals.end(),
            [](vector<int>& left, vector<int>& right) {
                return left[1] < right[1];
            }
        );
        
        int discarded = 0;
        int last = INT_MIN;
        for (vector<int> interval : intervals) {
            if (last > interval[0]) {
                ++discarded;
            } else {
                last = interval[1];
            }
        }
        return discarded;
    }
};

File: #0435.non-overlapping-intervals.js
/**
 * @param {number[][]} intervals
 * @return {number}
 */
var eraseOverlapIntervals = function(intervals) {
    intervals.sort((a, b) => a[0] - b[0]);
    let result = 0;
    for (let i = 1; i < intervals.length; i++) {
        if (intervals[i - 1][1] > intervals[i][0]) {
            result += 1;
            intervals[i][1] = Math.min(intervals[i][1], intervals[i - 1][1]);
        }
    }
    return result;
};
